gcc/d/d-objfile.cc

```diff
@@ -1694,11 +1696,6 @@ d_finish_symbol (tree decl)
       if (DECL_INITIAL (decl) == NULL_TREE)
 	{
 	  tree sinit = dtvector_to_tree (DECL_LANG_INITIAL (decl));
-	  if (TREE_TYPE (decl) == unknown_type_node)
-	    {
-	      TREE_TYPE (decl) = TREE_TYPE (sinit);
-	      TYPE_NAME (TREE_TYPE (decl)) = DECL_ASSEMBLER_NAME (decl);
-	    }
```

Comment:
Ultimately, this is what is being fixed here.  No more fake types generated by the codegen.